File: javascript/src/types/volumes.ts
export interface VolumeInfo {
  volume_id: string;
  tenant_id?: string;
  safety_identifier?: string;
  name?: string;
  oss_prefix?: string;
  bytes_used?: number;
  object_count?: number;
  usage_reconciled_at_unix?: number;
  created_at_unix?: number;
  updated_at_unix?: number;
}

export interface VolumeEntry {
  path: string;
  is_dir: boolean;
  size: number;
  modified_at_unix?: number;
}

export interface ListVolumesParams {
  limit?: number;
  page_token?: string;
  safety_identifier?: string;
}

export interface ListVolumesResponse {
  object: "list";
  data: VolumeInfo[];
  next_page_token?: string;
}

export interface CreateVolumeParams {
  name?: string;
  safety_identifier?: string;
}

export interface UpdateVolumeParams {
  name?: string;
  safety_identifier?: string;
  new_safety_identifier?: string;
}

export interface ListVolumeEntriesParams {
  path?: string;
  limit?: number;
  page_token?: string;
  safety_identifier?: string;
}

export interface SearchVolumeEntriesParams extends ListVolumeEntriesParams {
  query: string;
}

export interface ListVolumeEntriesResponse {
  object: "list";
  entries: VolumeEntry[];
  next_page_token?: string;
}

export interface ReadVolumeFileParams {
  max_bytes?: number;
  safety_identifier?: string;
}

export interface ReadVolumeFileRawParams {
  max_bytes?: number;
  format: "raw";
  safety_identifier?: string;
}

export type VolumeFileEncoding = "text" | "extracted_text" | "url" | "base64";

export interface VolumeFileDeliver {
  path: string;
  encoding: VolumeFileEncoding;
  mime_type: string;
  size: number;
  truncated: boolean;
  content?: string;
  content_base64?: string;
  image_url?: string;
  expires_at_unix?: number;
  extraction_warnings?: string[];
}

export interface VolumeFileRaw {
  path: string;
  size: number;
  truncated: boolean;
  content: ArrayBuffer;
  content_type?: string;
}

export interface VolumeFileWrite {
  path: string;
  size: number;
}

export interface VolumePathDelete {
  path: string;
  recursive: boolean;
}

export interface DownloadVolumeArchiveParams {
  path?: string;
  safety_identifier?: string;
}

export interface VolumeArchive {
  path: string;
  content: ArrayBuffer;
  content_type?: string;
}

export interface SummarizeVolumeParams {
  path?: string;
  safety_identifier?: string;
}

export interface VolumeSummaryPreview {
  path: string;
  size: number;
  preview: string;
  preview_truncated?: boolean;
}

export interface VolumeSummary {
  summary_path: string;
  file_count: number;
  total_bytes: number;
  top_paths_by_size: string[];
  text_previews: VolumeSummaryPreview[];
  generated_at_unix: number;
}

export interface ReadVolumeFileLinesParams {
  start_line: number;
  end_line?: number;
  max_bytes?: number;
  safety_identifier?: string;
}

export interface VolumeFileLines {
  path: string;
  start_line: number;
  end_line: number;
  total_lines: number;
  lines: string[];
  file_truncated: boolean;
  size: number;
}

export interface PatchVolumeFileLinesParams {
  start_line: number;
  end_line?: number;
  replacement?: string;
  safety_identifier?: string;
}

export interface VolumeFileLinesPatch {
  path: string;
  start_line: number;
  end_line: number;
  total_lines: number;
  size: number;
}

export interface GrepVolumeParams {
  pattern: string;
  path?: string;
  limit?: number;
  page_token?: string;
  safety_identifier?: string;
}

export interface VolumeGrepMatch {
  path: string;
  line_number: number;
  line: string;
}

export interface VolumeGrepResponse {
  object: "list";
  matches: VolumeGrepMatch[];
  next_page_token?: string;
  files_scanned: number;
  scan_truncated: boolean;
}

File: python/src/agent_api/types/volumes.py
from __future__ import annotations

from typing import Literal, TypedDict


class Volume(TypedDict, total=False):
    volume_id: str
    tenant_id: str
    name: str
    oss_prefix: str
    bytes_used: int
    object_count: int
    usage_reconciled_at_unix: int
    created_at_unix: int
    updated_at_unix: int


class VolumeEntry(TypedDict, total=False):
    path: str
    is_dir: bool
    size: int
    modified_at_unix: int


class ListVolumesResponse(TypedDict, total=False):
    object: Literal["list"]
    data: list[Volume]
    next_page_token: str


class ListVolumeEntriesResponse(TypedDict, total=False):
    object: Literal["list"]
    entries: list[VolumeEntry]
    next_page_token: str


class VolumeFileDeliver(TypedDict, total=False):
    path: str
    encoding: Literal["text", "extracted_text", "url", "base64"]
    mime_type: str
    size: int
    truncated: bool
    content: str
    content_base64: str
    image_url: str
    expires_at_unix: int
    extraction_warnings: list[str]


class VolumeFileRaw(TypedDict, total=False):
    path: str
    size: int
    truncated: bool
    content: bytes
    content_type: str


class VolumeFileWrite(TypedDict, total=False):
    path: str
    size: int


class VolumePathDelete(TypedDict, total=False):
    path: str
    recursive: bool


class VolumeFileLines(TypedDict, total=False):
    path: str
    start_line: int
    end_line: int
    total_lines: int
    lines: list[str]
    file_truncated: bool
    size: int


class VolumeFileLinesPatch(TypedDict, total=False):
    path: str
    start_line: int
    end_line: int
    total_lines: int
    size: int


class VolumeSummaryPreview(TypedDict, total=False):
    path: str
    size: int
    preview: str
    preview_truncated: bool


class VolumeSummary(TypedDict, total=False):
    summary_path: str
    file_count: int
    total_bytes: int
    top_paths_by_size: list[str]
    text_previews: list[VolumeSummaryPreview]
    generated_at_unix: int


class VolumeGrepMatch(TypedDict, total=False):
    path: str
    line_number: int
    line: str


class VolumeGrepResponse(TypedDict, total=False):
    object: Literal["list"]
    matches: list[VolumeGrepMatch]
    next_page_token: str
    files_scanned: int
    scan_truncated: bool


class VolumeArchive(TypedDict, total=False):
    path: str
    content: bytes
    content_type: str
